python/lance/_lib.pyx

```diff
@@ -248,10 +262,15 @@ def _lance_dataset_write(
     c_options.filesystem = filesystem.unwrap()
     c_options.file_write_options = write_options.unwrap()
     c_options.create_dir = True
-
-    if mode == "create":
-        c_mode = CLanceDataset.WriteMode.CREATE
-    elif mode == "append":
+    if max_rows_per_file > 0:
```

Comment:
let's add a simple check the max_rows_per_file >= max_rows_per_group?

Reply:
sgtm